packages/dcos-checks/extra/dcos-checks/cmd/components.go

```diff
@@ -0,0 +1,185 @@
+// Copyright © 2017 Mesosphere Inc. <http://mesosphere.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/dcos/dcos-go/dcos"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// master node has a 3dt instance running on TCP port 1050.
+	// ee version has 3dt running via unix socket on both master and agent nodes,
+	// depending on security option. Ports 80 or 443 are using accordingly.
+	dcosDiagnosticsMasterHTTPPort = 1050
+	adminrouterMasterHTTPSPort    = 443
+
+	// agent node runs 3dt via unix socket and is available though the agent
+	// adminrouter HTTP TCP port 61001 or HTTPS 61002.
+	adminrouterAgentHTTPPort  = 61001
+	adminrouterAgentHTTPSPort = 61002
+)
+
+var (
+	healthURLPrefix string
+)
+
+// componentsCmd represents the systemd health check
+var componentsCmd = &cobra.Command{
+	Use:   "components",
+	Short: "Check DC/OS components",
+	Long: `Check DC/OS components health by making a GET request to dcos-3dt service
+and validating the health field:
+
+/system/health/v1 is the local endpoint. The response structure is the following
+{
+  "units": ["unit1", ...]
+}
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// it is possible to implement a number of other checks under the same subcommand namespace.
+		// this function can be used as a dispatcher.
+		if len(args) == 0 {
+			RunCheck([]DCOSChecker{&ComponentCheck{
```

Comment:
Might be cleaner here to have another function 

```
func NewComponentCheck(name string) *ComponentCheck
```

I think that's what we've been talking about recently with regards to initializing structs.

Alternatively, you could define ComponentCheck as

```
type ComponentCheck string
```

And then just use `ComponentCheck("DC/OS components health check")`